### ISSUE TITLE
TextInput Component does not announce Checked/Unchecked State

### ISSUE THREAD

**Issue by amarlette** [facebook/react-native#30960]
## Description
Does not announce "checked"/"not checked" on focus of TextInput Component.

This issue is covered in-depth by parent issue #30843, please check there for more details.

## React Native version:
0.63

**Comment by martinivanalfonso** [comment]
I had a look at this issue and found out `accessible` & `accessibilityState` props are properly being passed down to the component (TouchableWithoutFeedback inside TextInput.js) however there seems to be some conflict with TextAncestor.Provider in the same file because when I removed it the screen reader was able to read all props including `checked`. 

This problem was not present on Button #30950  (which I can confirm was solved by PR #31001 ) as that component doesn't deal with an ancestor.

I'm not entirely sure how TextAncestor works so I can't debug any further unfortunately but I hope that bit helps

**Comment by fabriziobertoglio1987** [comment]
https://github.com/facebook/react-native/blob/051082117059e461f340f0fb7b0bde3db835e5dd/Libraries/Components/TextInput/AndroidTextInputNativeComponent.js#L549-L550

https://github.com/facebook/react-native/blob/051082117059e461f340f0fb7b0bde3db835e5dd/ReactCommon/react/renderer/components/textinput/androidtextinput/react/renderer/components/androidtextinput/AndroidTextInputShadowNode.cpp#L25

https://github.com/facebook/react-native/blob/051082117059e461f340f0fb7b0bde3db835e5dd/ReactCommon/react/renderer/components/textinput/androidtextinput/react/renderer/components/androidtextinput/AndroidTextInputProps.cpp#L133-L137

seems to me that we need to add `accessibilityState` in c plusplus component AndroidTextInputProps

**Comment by blavalla** [comment]
While technically this would be possible, I can't imagine a UI where this would be useful or necessary. Closing out.
